atlasdb-client/src/main/java/com/palantir/atlasdb/util/SlidingWindowMetricsInvocationHandler.java

```diff
@@ -41,12 +46,22 @@
     private static final Logger logger = LoggerFactory.getLogger(SlidingWindowMetricsInvocationHandler.class);
 
     private final MetricRegistry metricRegistry;
+    private final Map<Method, Timer> timers;
     private final String serviceName;
 
-    public SlidingWindowMetricsInvocationHandler(MetricRegistry metricRegistry, String serviceName) {
+    public SlidingWindowMetricsInvocationHandler(
+            MetricRegistry metricRegistry, String serviceName, Class<?> clazz) {
         super(InstrumentationUtils.getEnabledSupplier(serviceName));
         this.metricRegistry = Preconditions.checkNotNull(metricRegistry, "metricRegistry");
         this.serviceName = Preconditions.checkNotNull(serviceName, "serviceName");
+        this.timers = Arrays.stream(clazz.getDeclaredMethods())
+                .collect(ImmutableMap.toImmutableMap(Function.identity(), this::getTimer));
```

Comment:
You may want to do something along these lines to avoid expensive operations for methods inherited from super-interfaces: https://github.com/palantir/tritium/pull/358

Reply:
:+1: I will do this in a follow-up